### PR TITLE
Assembler: remove check icon from the category list

### PR DESCRIPTION
**Pull request by fushar**
Related to: p1698783267063269-slack-CRWCHQGUB

## Proposed Changes

Remove the check icon from selected category, to make the sidebar "less busy".

|Before|After|
|-|-|
|image|image|

cc: @lucasmendes-design we'll need your approval on this 🙏 

## Testing Instructions

1. Go to the Assembler flow.
2. Select some patterns.
3. Verify that the categories that have at least one pattern selected, retain their original icon.
4. Verify that you can complete flow as usual.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/onboarding/src/navigator/navigator-buttons/index.tsx [hunk 1]
```diff
@@ -22,10 +22,10 @@ interface NavigatorButtonAsItemProps extends NavigatorItemProps {
 	path: string;
 }
 
-export function NavigatorItem( { icon, checked, active, children, ...props }: NavigatorItemProps ) {
+export function NavigatorItem( { icon, active, children, ...props }: NavigatorItemProps ) {
```
Comment (arthur791004) [inline packages/onboarding/src/navigator/navigator-buttons/index.tsx]: We also need to remove the `checked` prop from the type definition

**Review by arthur791004 — COMMENTED** [review]
We need to avoid passing the `checked` prop from Header & Footer as well.

See https://github.com/Automattic/wp-calypso/blob/0b2a73d416d16d84b8efb9b218260de568ab6c4c/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx#L74

File: packages/onboarding/src/navigator/navigator-buttons/index.tsx (hunk 1 above)
Comment (fushar) [reply packages/onboarding/src/navigator/navigator-buttons/index.tsx]: We cannot, since the check icon is still required in the `Select styles` screen.

**Review by taipeicoder — APPROVED** [review]
LGTM! 👍

**Comment by lucasmendes-design** [comment]
I tested and it's perfect for me :) 

Let's compare the before and after. We've seen a significant improvement in pattern selection ( https://wp.me/pekYwv-34D#comment-2555 ), and I wonder if the checkmark guides users or nudges users to select more patterns. I'm not sure, probably, this will not make any impact but I think we can check that :)

**Pull request merged**